recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
+        )
+
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        autotools = self._configure_autotools()
-        if self.settings.build_type == "Debug":
-            autotools.install(args=self._make_args)
-        else:
-            autotools.make(args=["install-strip"] + self._make_args)
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.remove_files_by_mask(self.package_folder, "*.la")
-        self.copy(pattern="COPYING*", dst="licenses", src=self._source_subfolder)
+        autotools = Autotools(self)
+        autotools.make(target="install-strip")
+
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rm(self, "*.la", self.package_folder, recursive=True)
+        copy(
+            self,
+            pattern="COPYING*",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+            keep_path=False,
+        )
 
     def package_info(self):
-        bindir = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : " + bindir)
-        self.env_info.PATH.append(bindir)
+        if self.info.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.append("m")
+            self.cpp_info.system_libs.append("rt")
+            self.cpp_info.system_libs.append("pthread")
+            self.cpp_info.system_libs.append("dl")
 
-        cc = os.path.join(bindir, "gcc-%s" % self.version)
-        self.output.info("Creating CC env var with : " + cc)
-        self.env_info.CC = cc
+        bindir = os.path.join(self.package_folder, "bin")
 
-        cxx = os.path.join(bindir, "g++-%s" % self.version)
-        self.output.info("Creating CXX env var with : " + cxx)
-        self.env_info.CXX = cxx
+        cc = os.path.join(bindir, f"gcc-{self.version}")
+        self.output.info("Creating CC env var with: " + cc)
+        self.buildenv_info.define("CC", cc)
+
+        cxx = os.path.join(bindir, f"g++-{self.version}")
+        self.output.info("Creating CXX env var with: " + cxx)
+        self.buildenv_info.define("CXX", cxx)
+
+        fc = os.path.join(bindir, f"gfortran-{self.version}")
+        self.output.info("Creating FC env var with: " + fc)
+        self.buildenv_info.define("FC", fc)
+
+        ar = os.path.join(bindir, f"gcc-ar-{self.version}")
+        self.output.info("Creating AR env var with: " + ar)
+        self.buildenv_info.define("AR", ar)
+
+        nm = os.path.join(bindir, f"gcc-nm-{self.version}")
+        self.output.info("Creating NM env var with: " + nm)
+        self.buildenv_info.define("NM", nm)
+
+        ranlib = os.path.join(bindir, f"gcc-ranlib-{self.version}")
+        self.output.info("Creating RANLIB env var with: " + ranlib)
+        self.buildenv_info.define("RANLIB", ranlib)
+
+        # TODO: Remove after conan 2.0 is released
+        if Version(conan_version).major < 2:
```

Comment:
We are not allowing the version check.

We want to ensure a smooth migration and the best way for us is to mesure the v2 pipeline automatically and these hide problems the client should have a fix for this in the next beta release

Reply:
Ah, no worries. Can you suggest an alternative method of passing these variables through to the v1 generators, since they won't have access to `self.{build,run}env_info`? Without this section, `test_v1_package` will fail. I was referring to the `emsdk` recipe for inspiration for this solution: https://github.com/conan-io/conan-center-index/blob/53ef0688c436c6697d8312d47fcdef68fe7e6955/recipes/emsdk/all/conanfile.py#L179

Reply:
Keep without that check, that's it. When Conan v2 become the standard version, we will need update all recipes anyway.